modin/experimental/engines/omnisci_on_native/io.py

```diff
@@ -245,82 +244,14 @@ def read_csv(
-            # columns unless it's explicitly requested via the 'parse_dates' parameter.
-            # Aligning these differences requires passing explicit type-scheme to Arrow
-            # which includes proper types for datetime-like columns.
-            if parse_dates is not None:
-                # Deducing columns which potentially have a timestamp data-type to prohibit for
-                # Arrow to infer their types by explicitly setting them to string.
-                # TODO: potentially do this only in a strict compatibility mode as this may
-                # really harm performance for wide frames
-                # if STRICT_COMPATIBILITY:
-                datetime_columns = (
-                    pandas.read_csv(
-                        **dict(
-                            mykwargs,
-                            nrows=1,
-                            parse_dates=column_names,
-                            dtype=None,
-                            engine=None if eng == "arrow" else engine,
-                        ),
-                    )
-                    .select_dtypes("datetime")
-                    .columns
-                )
+                column_types = cls._dtype_to_arrow(dtype)
 
-            if is_list_like(parse_dates):
-                parse_dates_labels = frozenset(
-                    column_names[col] if isinstance(col, int) else col
-                    for col in parse_dates
-                )
-                for col in datetime_columns:
-                    if col not in parse_dates_labels:
-                        column_types[col] = pa.string()
-
-                for col in parse_dates_labels:
-                    # 'parse_dates' parameter means to try to parse specific columns into timestamps,
-                    # if it couldn't be done treat this column as a string. Setting explicit timestamp
-                    # type for such column will trigger a parse error in Arrow if the column content
-                    # is not a date/time. Deleting such columns from 'column_types' allows Arrow to
-                    # infer dtypes (treat as timestamp if it's possible or as a string otherwise)
-                    # and so behave like pandas.
-                    column_types.pop(col, None)
-            elif isinstance(parse_dates, bool):
-                index_col_labels = ()
-                if index_col is not None:
-                    index_col = index_col if is_list_like(index_col) else (index_col,)
-                    index_col_labels = frozenset(
-                        column_names[col] if isinstance(col, int) else col
-                        for col in index_col
-                    )
-                for col in datetime_columns:
-                    if not (parse_dates and col in index_col_labels):
-                        column_types[col] = pa.string()
+            if (type(parse_dates) is list) and type(column_types) is dict:
```

Comment:
Why don't you do `if isinstance(parse_dates, list) and isinstance(column_types, dict)`?
Also why exact type check instead of duck typing?..

Reply:
this commit just reverts `read_csv` to the state it's presented in the current master. Do we want to do the suggested changes in this PR?